tests/neg/i17192.5.scala

```diff
@@ -0,0 +1,12 @@
+class Ifce[BT <: Boolean]:
+  type RT = BT match
+    case true => this.type { val v1: Int }
```

Comment:
Add
```suggestion
  type RT = BT match
    case true  => this.type { val v1: Int }
    case false => this.type
```
to be exactly the same situation as the pos test, modulo the `extends Selectable`?